src/daemon/daemon.cpp

```diff
@@ -1083,81 +1083,126 @@ void mp::Daemon::find(const FindRequest* request, grpc::ServerWriterInterface<Fi
+
+                for (const auto& [remote, info] : vm_images_info)
+                {
+                    std::string name;
+                    if (info.aliases.contains(QString::fromStdString(request->search_string())))
+                    {
+                        name = request->search_string();
+                    }
+                    else
+                    {
+                        name = info.id.toStdString();
+                        name.resize(12);
+                    }
+
+                    auto entry = response.add_images_info();
+                    entry->set_os(info.os.toStdString());
+                    entry->set_release(info.release_title.toStdString());
+                    entry->set_version(info.version.toStdString());
+                    auto alias_entry = entry->add_aliases_info();
+                    if (!request->remote_name().empty() ||
+                        (request->remote_name().empty() && vm_images_info.size() > 1 && remote != default_remote))
+                    {
+                        alias_entry->set_remote_name(remote);
+                    }
+                    alias_entry->set_alias(name);
+                }
             }
             catch (const std::exception&)
             {
-                throw std::runtime_error(
-                    fmt::format("Unable to find an image or Blueprint matching \"{}\"", request->search_string()));
+                // Ignore exception, no images found
             }
         }
 
-        for (const auto& [remote, info] : vm_images_info)
+        if (request->show_blueprints())
         {
-            std::string name;
-            if (info.aliases.contains(QString::fromStdString(request->search_string())))
-            {
-                name = request->search_string();
-            }
-            else
+            try
             {
-                name = info.id.toStdString();
-                name.resize(12);
-            }
+                VMImageInfo info = config->blueprint_provider->info_for(request->search_string());
 
-            auto entry = response.add_images_info();
-            entry->set_os(info.os.toStdString());
-            entry->set_release(info.release_title.toStdString());
-            entry->set_version(info.version.toStdString());
-            auto alias_entry = entry->add_aliases_info();
-            if (!request->remote_name().empty() ||
-                (request->remote_name().empty() && vm_images_info.size() > 1 && remote != default_remote))
+                std::string name;
+                if (info.aliases.contains(QString::fromStdString(request->search_string())))
+                {
+                    name = request->search_string();
+                }
+                else
+                {
+                    name = info.id.toStdString();
+                    name.resize(12);
+                }
+
+                auto entry = response.add_blueprints_info();
```

Comment:
I wanted to change the `add_aliases()` function to take either `response.add_images_info()` or `response.add_blueprints_info()` with the use of templates, but wasn't able to make it work resulting in this very similar code. eg.
```
template<typename T>
void add_aliases(T entry, const std::string& remote_name, const mp::VMImageInfo& info,
                 const std::string& default_remote)
```
Any suggestions?